src/Periphery/TokenWrapper.sol

```diff
@@ -50,10 +72,22 @@ contract TokenWrapper is WithdrawablePeriphery {
         // to have `wad` equal to the minimum between the balance and the
         // given allowance, in our specific usecase allowance is always
         // nearly MAX_UINT256. Using the balance only is a gas optimisation.
-        uint256 wad = IERC20(wrappedToken).balanceOf(msg.sender);
-        IERC20(wrappedToken).transferFrom(msg.sender, address(this), wad);
-        IWrapper(wrappedToken).withdraw(wad);
-        SafeTransferLib.safeTransferETH(msg.sender, wad);
+        uint256 wad = IERC20(WRAPPED_TOKEN).balanceOf(msg.sender);
+        IERC20(WRAPPED_TOKEN).transferFrom(msg.sender, address(this), wad);
+
+        if (USE_CONVERTER) {
+            // Approve converter to spend wrappedToken
+            LibAsset.maxApproveERC20(IERC20(WRAPPED_TOKEN), CONVERTER, wad);
```

Comment:
I think this should be moved into the constructor to do it once (and for all) and save gas on all subsequent calls